MetaMask/metamask-extension
# Update Babel and dependencies

## Explanation

This pull request updates our babel and its dependencies. The goal here is to unlock further build system improvements as well as tap into improvements made to babel as well as unlock TypeScript 4.7 support. 

Things to note:
1. A number of plugins, such as object rest spread, nullish coalescing and optional chaining are not part of preset-env and are no longer needed explicitly. 
2. The 'targets' property in `babel.config.js` is moved to the top level scope instead of preset-env.
3.  Babel runtime was updated and patched in a similar fashion as the previous version.

## More Information
I am keeping a close eye on the MV3 metrics in relation to this PR. Once i'm solid i'm going to ask for @jpuri to do an analysis of the impact. 

TODO:
- [x] Confirm that adding/removing babel-runtime plugin does not impact bundle size / load times
- [x] Get sign off from MV3 team

## Manual Testing Steps
1. Load the extension
2. Check out functionality in general
3. All E2E tests pass so asking for manual review just to make sure nothing crazy happened.



## Pre-Merge Checklist

- ~**IF** this PR fixes a bug, a test that _would have_ caught the bug has been added~
- ~PR is linked to the appropriate GitHub issue~

### + If there are functional changes:
